### ISSUE TITLE
Product Filter Price (Beta) Block: Missing Currency Sign in Input Fields

### ISSUE THREAD

**Issue by thealexandrelara** [woothemes/woocommerce#45315]
### Describe the bug

In the Product Filter Price (Beta) block, the currency sign is expected to be visible within the price input fields to provide clear context on the currency being used. However, it appears that the currency sign does not show up inside the input fields, which can lead to confusion for users attempting to set price filters based on their local currency. 

This issue was raised by @kmanijak in https://github.com/woocommerce/woocommerce/pull/45197#pullrequestreview-1916834755

### Expected behavior

The currency sign should be displayed inside the price input fields. 

### Actual behavior

The currency sign is not shown in the price input fields within the Product Filter Price (Beta) block. Users see only numerical values without any currency indication.

### Steps to reproduce

1. Navigate to Single Product template (Appearance > Editor > Templates > Product Catalog template - or any archive template).
2. Click on "Edit" to open the WordPress editor.
3. In the editor, you should see a block inserter icon or an option to "Add Block". Click on it.
4. In the block inserter, search for "Product Filter" or browse the available blocks until you find them.
5. Click on the "Product Filter: Price (Beta)" block to add it to your content.
6. Check that no currency is displayed inside the price input fields.
7. Click on Save to save the changes.
8. Visit the Shop page (https://linktoyourstore.com/shop/).
9. Check that no currency is displayed inside the price input fields.

**Comment by dinhtungdu** [comment]
Can we also consider accepting the lack of the currency sign? I know that that React-based version has the sign but it's quite complex to do the same in HTML. cc @jarekmorawski for design input.

**Comment by thealexandrelara** [comment]
> Can we also consider accepting the lack of the currency sign? I know that that React-based version has the sign but it's quite complex to do the same in HTML. cc @jarekmorawski for design input.

Since we are using Interactivity API, I wonder if we couldn't just use a `selector` with the formatted price. This way, we would be able to manipulate the price on the JavaScript side. However, let's wait for feedback from @jarekmorawski before exploring this possibility.

**Comment by dinhtungdu** [comment]
@thealexandrelara Yeah, we can use the derived state to display the formatted price. But we also need to validate the price and extract the amount from the input, which makes the block more complex. I'm comparing the trade-off between two options and wonder if it's worth the effort.

**Comment by kmanijak** [comment]
> But we also need to validate the price and extract the amount from the input, which makes the block more complex. I'm comparing the trade-off between two options and wonder if it's worth the effort.

You can check the price input experience in new Product Editor or in Product Collection price range filter (PR by @imanish003: https://github.com/woocommerce/woocommerce/pull/42858). You can reuse[ the logic from here](plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/price-range-control/PriceTextField.tsx).

It's not my decision to take, but I have a pretty strong opinion that a currency sign is a must. Personally, as a buyer, I'd be even suspicious if the shop is legit if it doesn't display currency in a currency input (even if it's just a filter).

**Comment by dinhtungdu** [comment]
@kmanijak thanks for the suggestions. I'm not against adding that too, I'm just considering available options. 

Looking at the price range filter, I notice that if I enter a price with the currency sign in different formats than the one set in the WooCommerce setting, then the price is cleared out after the move focus out of the input, does this issue under your radar?

https://github.com/woocommerce/woocommerce/assets/5423135/56b4aeed-ccf7-437c-90f2-ff32a0401d3a

**Comment by kmanijak** [comment]
> Looking at the price range filter, I notice that if I enter a price with the currency sign in different formats than the one set in the WooCommerce setting, then the price is cleared out after the move focus out of the input, does this issue under your radar?

Oops, something weird's happening there 😅 cc: @imanish003

**Comment by jarekmorawski** [comment]
Do you still need my input? From the shopper's POV, the currency is important, so I'd prefer to do everything in our power to display it on the front-end. If it's incredibly difficult, maybe we can make it simpler by displaying the symbol outside the input fields.

**Comment by dinhtungdu** [comment]
@jarekmorawski Thanks for your input here! 

> From the shopper's POV, the currency is important, so I'd prefer to do everything in our power to display it on the front-end.

Yep, I agree, it's also not incredibly difficult 😄, so let's do this one cc @thealexandrelara.

**Comment by imanish003** [comment]
@dinhtungdu AFAIU, you are manually adding/copying a different currency sign into input, Right? If yes, we are clearing the input because it is not valid.

- First, the user doesn't need to add a currency sign manually. When the user enters the value, it will automatically be added as a prefix.
- Second, even if the user decides to add the currency sign manually, I think the merchant should only enter the Store's currency. In this case, the bug won't happen.

**Question:** How do you think it should behave?
- Should we replace the manually entered (different) currency with store currency?

**Comment by dinhtungdu** [comment]
> AFAIU, you are manually adding/copying a different currency sign into input, Right?

@imanish003 No. In the video above:

1. I first type only `2`.
2. The value is then replaced by `2 $`. The store currency is `$`, with a space between value and sign.
3. I updated the price to 20 by moving the cursor next to `2` and then adding `0`, so my input is `20 $`. 
4. The input is still work as expected at this point.
5. I then removed the space between `20` and `$`, my input now is `20$` => the input is clear after submitting.

In other words, I type the price with the correct currency sign, but in a slightly different format (missing a space). The input treats that price as invalid for now. I don't think we should be so strict like that, because space can be missed easily.

> How do you think it should behave?

IMO, like what I did in https://github.com/woocommerce/woocommerce/pull/45409, we can extract only the number and reformat the price again for display. I don't think merchants should enter a price with a different currency than what the store is using, but they can accidentally type other characters, like hitting the dash near the zero key. Instead of clearing everything, we can clear only the irrelevant parts like the `-` and keep the number.
